ctrd/client.go

```diff
@@ -231,9 +231,14 @@ func (c *Client) collectContainerdEvents() {
 				logrus.Warnf("failed to parse %s event: %#v", TaskExitEventTopic, out)
 				continue
 			}
-			action = "die"
+			if exitEvent.ID == exitEvent.ContainerID {
+				action = "die"
+			} else {
+				action = "exec_die"
```

Comment:
In swagger.yml, we defined the /events API like the following:
```
  /events:
    get:
      summary: "Subscribe pouchd events to users"
      description: |
        Stream real-time events from the server.
        Report various object events of pouchd when something happens to them.
        Containers report these events: create`, `destroy`, `die`, `oom`, `pause`, `rename`, `resize`, `restart`, `start`, `stop`, `top`, `unpause`, and `update`
        Images report these events: `pull`, `untag`
        Volumes report these events: `create`, `destroy`
        Networks report these events: `create`, `connect`, `disconnect`, `destroy`
```
But here, I a afraid that we are adding another event type for container : exec's exec_die, right?
While this has not been collected in to API illustration. If we need to add this in swagger.yml, we should fix that. @HusterWan 
And maybe we could have a very thorough thinking for the all workflow of the change: API, daemon, containerd, doc, test and some others.

Reply:
great advise, i will update this or soon